.github/workflows/publish.yaml

```diff
@@ -0,0 +1,75 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      -
+        name: Docker meta
+        id: meta
+        uses: crazy-max/ghaction-docker-meta@v1
```

Comment:
@alexellis to answer you question about how this differ from classic watchdogt is only here in how the docker metatdata is generated versus https://github.com/openfaas/classic-watchdog/blob/f5f0832493e8165695673fa6855039e08ef524a6/.github/workflows/publish.yaml#L25-L27

Reply:
Do you see any issue in replicating the approach from the classic watchdog here, so that they are the same and easier to maintain?

Reply:
I guess, originally this was my recommendation on what we _should_ be doing. But this PR was ignored and the current pattern adopted without considering this one. I will update it to match

Reply:
Thanks that would be appreciated. There is a rebase needed too, for the Dockerfile.

Reply:
@alexellis after making the changes i remember the main benefit of this action for creating the tag, this step understands your forked repo and creates the tag accordingly. The workflow for classic-watchdog does not handle forks correctly and can not push to the forked ghcr because the repository is fixed to `openfaas`

if you are ok with not having it work in forks, then i guess we can remove this, but if you want forked image builds to actually be publishable automatically, we would either need to add more complexity to the workflow _or_ use this action

Reply:
a note about using this action, it is completely templated, so the exact same snippet can be used in _all_ repos, versus  the currently hard coded repo name that needs to be updated to match the repo you copy it to

Reply:
Templates make sense to me, that is something both should have IMHO.

My real concern is having two divergent builds for the two watchdogs, which should both be doing the same thing

build go binary and upload it
build docker image and upload it

Reply:
they both do the same flow now, I rebased and updated to copy all of the workflows and makefile changes from classic-watchdog 

but I think that `crazy-max/ghaction-docker-meta@v1` is probably a more robust way to construct the docker tags and labels